### ISSUE TITLE
Improve uprops data pipeline

### ISSUE THREAD

**Issue by iainireland** [unicode-org/icu4x#936]
Following up on the minimal uprops provider work in issue #882 and issue #883, we should:

1. Add support in `icu4x-datagen` (somewhat complicated by the fact that compiling and running an ICU4C tool is necessary)
2. Add example in `icu_uniset::props`

**Comment by sffc** [comment]
> 1. Add support in `icu4x-datagen` (somewhat complicated by the fact that compiling and running an ICU4C tool is necessary)

No, we should not depend on ICU4C being run.  The idea is that ICU4C ships a zip file with all the properties pre-exported.  For the time being, `icu4x-datagen` should take a path to that directory, with instructions on how to generate it manually, and eventually (later this quarter or in Q4) you will be able to download that directory as a zip file from GitHub.

**Comment by iainireland** [comment]
Zip file export is being added to ICU4C here: https://github.com/unicode-org/icu/pull/1857.
